python/activator/activator.py

```diff
@@ -554,14 +554,31 @@ def keda_start():
                         # Process fan out visit
                         process_visit(expected_visit)
                         processing_result = "Success"
-                    except GracefulShutdownInterrupt:
-                        _log.error(
-                            "Service interrupted.Shutting down *without* syncing to the central repo.")
-                        processing_result = "Interrupted"
-                        sys.exit(1)
                     except IgnorableVisit as e:
                         _log.info("Skipping visit: %s", e)
                         processing_result = "Ignore"
+                    except GracefulShutdownInterrupt:
+                        # Safety net to minimize chance of interrupt propagating out of the worker.
+                        _log.error("Service interrupted.")
+                        processing_result = "Interrupted"
+                        sys.exit(1)
+                    except RetriableError as e:
```

Comment:
I guess given the TODO that these cases will diverge at some point?  Otherwise, `except RetriableError, NonRetriableError as e:` would save some duplication.

Reply:
Yes, that's why they're different exceptions in the first place. An interrupt's recoverable _a priori_, but if it happens after APDB updates...